Formula/knot-resolver.rb

```diff
@@ -0,0 +1,101 @@
+class KnotResolver < Formula
+  desc "Minimalistic, caching, DNSSEC-validating DNS resolver"
+  homepage "https://www.knot-resolver.cz"
+  url "https://secure.nic.cz/files/knot-resolver/knot-resolver-1.2.1.tar.xz"
+  sha256 "1b6f55ea1dfec90f45c437f23e1ab440e478570498161d0f8a8f94a439305f8c"
+  revision 1
+
+  head do
+    url "https://gitlab.labs.nic.cz/knot/resolver.git"
+  end
+
+  depends_on "knot"
+  depends_on "luajit"
+  depends_on "libuv"
+  depends_on "gnutls"
+  depends_on "lmdb"
+
+  depends_on "cmocka" => :build
+  depends_on "pkg-config" => :run
+
+  option "without-nettle", "Compile without DNS cookies support"
+  option "with-hiredis", "Compile with Redis cache storage support"
+  option "with-libmemcached", "Compile with memcached cache storage support"
+  depends_on "nettle" => :recommended
+  depends_on "hiredis" => :optional
+  depends_on "libmemcached" => :optional
+
+  def install
+    %w[all check lib-install daemon-install modules-install].each do |target|
+      system "make", target, "PREFIX=#{prefix}"
+    end
+
+    (buildpath/"config").write(config)
+    (etc/"kresd").install "config"
+
+    (buildpath/"root.keys").write(root_keys)
+    (var/"kresd").install "root.keys"
+  end
+
+  def config; <<-EOS.undent
+    -- vim:syntax=lua:
+    -- Refer to manual: http://knot-resolver.readthedocs.org/en/latest/daemon.html#configuration
+
+    net.listen(net.lo0)
+
+    -- Unmanaged DNSSEC root TA
+    trust_anchors.file = 'root.keys'
+
+    -- Load useful modules
+    modules = {
+        'stats',    -- Track internal statistics
+        'policy',   -- Block queries to local zones/bad sites
+        'predict',  -- Prefetch expiring/frequent records
+    }
+
+    -- Cache size
+    cache.size = 100 * MB
+    EOS
+  end
+
+  # DNSSEC root anchor published by IANA
+  def root_keys; <<-EOS.undent
```

Comment:
These files are not strictly configuration files. It's a secure trust anchor for DNSSEC (similarly to CA certificates in PKI). And it's kind of a chicken-and-egg problem. We need to provide the anchor securely.

Most Linux distributions decided to do so in package form. The trust is already established towards the distribution and the integrity of the package can be verified using GPG signatures for instance. There is either a single package that contains the anchor and other DNSSEC-aware validators can use it (Debian). Or the anchor is shipped with each component that needs it (Fedora).

Bootstrapping via HTTPS relies on certificate authorities. Some consider this concept broken because CAs are difficult to audit. In contrast a distribution represent a single point of trust. Knot Resolver can use HTTPS for bootstrap and will also authenticate the downloaded file using out-of-band X.509, however that requires us to include ICANN public certificate instead.

I don't mind removing the trust anchor from the formula. But it makes sense to me to include it this way, because it means less moving parts and allows the users to use DNSSEC validation immediately.